python/meterpreter/meterpreter.py

```diff
@@ -713,9 +733,21 @@ def encrypt_packet(self, pkt):
         # The packet now has to contain session GUID and encryption flag info
         # And given that we're not yet supporting AES, we're going to just
         # always return the session guid and the encryption flag set to 0
-        # TODO: we'll add encryption soon!
+        enc_type = ENC_NONE
+        if self.aes_key:
+            # We've got a key, but only encrypt if it's enabled
+            if self.aes_enabled:
+                iv = rand_bytes(16)
+                enc = iv + met_aes_encrypt(self.aes_key, iv, pkt[8:])
+                hdr = struct.pack('>I', len(enc) + 8) + pkt[4:8]
+                pkt = hdr + enc
+                enc_type = ENC_AES256
+            else:
+                # We enable it here.
```

Comment:
From what I can tell, it's enabled here because it is set but not enabled for the response to the `negotiate_tlv_encryption` request and after that it is used.

I might update the comment here to clarify that for future readers. At first glance it's not clear why it would be set but disabled and then flipped to enabled here.

Reply:
Should be good now mate. please check!